### PR TITLE
[bitnami/airflow]: add servicemonitor for metrics scraping fixes #6552

### LINKED ISSUE #6552: [bitnami/airflow] add ServiceMonitor if servicemonitor enabled true to automatically scrape exporter metrics
**Which chart**:
airflow all versions

**Is your feature request related to a problem? Please describe.**
Metrics exporting

**Describe the solution you'd like**
add a flag in values.yaml metrics.serviceMonitor: true/false, if enabled we will create the following manifest
```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: airflow-metrics
spec:
  endpoints:
  - port: metrics
  selector:
    matchLabels:
      name: airflow-exporter
```

**Describe alternatives you've considered**
I created the above servicemonitor manually

### PR DESCRIPTION
**Pull request by iamsudip**
**Description of the change**

Adds ServiceMonitor

**Benefits**

Enabled prometheus scraping if the prometheus is managed using prometheus operator

**Possible drawbacks**
NA

**Applicable issues**

  - fixes #6552

**Additional information**

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/airflow/templates/metrics/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{{- if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "common.names.fullname" . }}-metrics
```
Comment (Mauraza) [inline bitnami/airflow/templates/metrics/servicemonitor.yaml:5]: Could you add the namespace too? something like this:
```
{{- if .Values.metrics.serviceMonitor.namespace }}
  namespace: {{ .Values.metrics.serviceMonitor.namespace | quote }}
  {{- else }}
  namespace: {{ .Release.Namespace | quote }}
  {{- end }}
```

File: bitnami/airflow/Chart.yaml [hunk 2]
```diff
@@ -32,4 +32,4 @@ name: airflow
 sources:
   - https://github.com/bitnami/bitnami-docker-airflow
   - https://airflow.apache.org/
-version: 10.0.8
+version: 10.0.9
```
Comment (Mauraza) [inline bitnami/airflow/Chart.yaml:35]: could you change the version update to `10.1.0`?

**Review by Mauraza — CHANGES_REQUESTED** [review]
Hi @iamsudip,

Thanks for the PR!!, could you check the comments?

File: bitnami/airflow/values.yaml [hunk 3]
```diff
@@ -970,6 +970,14 @@ metrics:
...
   hostAliases: []
 
+  ## Prometheus Operator ServiceMonitor resource
+  serviceMonitor:
+    enabled: false
+    interval: 60s
+    scrapeTimeout: 10s
+    # ServiceMonitor Labels
+    labels: {}
+
```
Comment (Mauraza) [inline bitnami/airflow/values.yaml:980]: sorry I missed this. Could you add the namespace in values.yaml?
```
## Namespace in which Prometheus is running
    ##
    # namespace: monitoring
    ## Interval at which metrics should be scraped.
    ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#endpoint
```

**Review by Mauraza — COMMENTED** [review]
Is almost done!

File: bitnami/airflow/values.yaml (hunk 3 above)
Comment (sudip-moengage) [reply bitnami/airflow/values.yaml:984]: @Mauraza Can you check now?

Comment (iamsudip) [reply bitnami/airflow/values.yaml:984]: @Mauraza can you check now?

**Review by Mauraza — APPROVED** [review]
Hi @iamsudip,

Thanks again for the PR! LGTM!!

**Pull request merged**